docs/resources/vod_transcoding_template_group.md

```diff
@@ -0,0 +1,145 @@
+---
+subcategory: "Video on Demand (VOD)"
+---
+
+# huaweicloud_vod_media_category
+
+Manages a VOD transcoding template group resource within HuaweiCloud.
+
+## Example Usage
+
+```hcl
+resource "huaweicloud_vod_transcoding_template_group" "test" {
+  name        = "test"
+  description = "test group"
+
+  common {
+    audio_codec          = "AAC"
+    hls_segment_duration = 5
+    low_bitrate_hd       = false
+    video_codec          = "H264"
+  }
+
+  quality_info {
+    output_format = "HLS"
+
+    audio {
+      bitrate     = 0
+      channels    = 2
+      sample_rate = 1
+    }
+
+    video {
+      bitrate    = 1000
+      frame_rate = 1
+      height     = 720
+      quality    = "HD"
+      width      = 1280
+    }
+  }
+}
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `region` - (Optional, String, ForceNew) Specifies the region in which to create the resource. If omitted, the
+  provider-level region will be used. Changing this creates a new resource.
+
+* `name` - (Required, String) Specifies the name of the template group. The value can be a string of 1 to 128
+  characters that can consist of letters, digits and underscores (_).
+
+* `description` - (Optional, String) Specifies the description of the template group.
+
+* `auto_encrypt` - (Optional, Bool) Specifies whether to automatically encrypt. Before enabling, you need to configure
+  the HLS encryption key URL. When `auto_encrypt` is **true**, the `output_format` must be **HLS**.
+  Defaults to: **false**.
+
+* `is_default` - (Optional, Bool) Specifies whether to use this group as default group. Defaults to: **false**.
+
+* `watermark_template_ids` - (Optional, List) Specifies the list of the watermark template IDs.
+
+* `common` - (Required, List) Specifies the common configurations of the template group.
+  The [object](#common_object) structure is documented below.
+
+* `quality_info`- (Required, List) Specifies the quality info list of the template group.
+  The [object](#quality_info_object) structure is documented below.
```

Comment:
pls bring the required parameters ahead of the optional parameters

Reply:
fixed